### PR TITLE
Added support for detecting changes in layer styles

### PR DESCRIPTION
**Pull request by theduckylittle**
1. Added support for detecting changes in a layers
   style.
2. Created a demo of changing the paint settings.
3. Small fixes to linting.

### CODE REVIEW

File: src/components/map.jsx [hunk 1]
```diff
@@ -191,6 +193,18 @@ function fakeStyle(layer) {
   }, layer.source);
 }
 
+/** Get a layer by it's id
+ */
+function getLayerById(layers, id) {
```
Comment (wnordmann) [inline src/components/map.jsx]: Can this be moved to the util.js

File: src/components/map.jsx [hunk 2]
```diff
@@ -392,6 +423,9 @@ export class Map extends React.Component {
         // move on with the loop if the layer
         //  could not be defined.
         if (layer_def === null) {
+          // WARNING! Continue is used to short circuit
+          //          the rest of this loop.
+          // eslint-disable-next-line no-continue
           continue;
```
Comment (wnordmann) [inline src/components/map.jsx]: could this avoided using an else?

File: src/components/map.jsx [hunk 3]
```diff
@@ -381,7 +412,7 @@ export class Map extends React.Component {
             // layersDef[x] will contain objects which need to be
```
Comment (wnordmann) [inline src/components/map.jsx]: I always like the convention of moving iterator variables down the alphabet, i, j, k, l, ...

File: examples/index.html [hunk 4]
```diff
@@ -26,6 +26,10 @@
         <a href="./clustering/index.html">Clustering Example</a><br>
         Demonstrates clustering points and changing radius.
     </li>
+    <li>
```
Comment (wnordmann) [inline examples/index.html]: I added this, I wanted it to make my review easier.

Comment (theduckylittle) [inline examples/index.html]: :+1:

File: src/components/map.jsx (hunk 1 above)
Comment (theduckylittle) [inline src/components/map.jsx]: Moved.

File: src/components/map.jsx (hunk 3 above)
Comment (theduckylittle) [inline src/components/map.jsx]: Renamed

File: src/components/map.jsx (hunk 2 above)
Comment (theduckylittle) [inline src/components/map.jsx]: Added a "layer !== null" into the if statements below it. Continue is removed.

**Pull request merged**